### ISSUE TITLE
Issue #24: fine-tuning dataset export を追加する

### ISSUE THREAD

**Issue by Kewton** [Kewton/Anvil#473]
## 概要

Reminder 専用モデルなどの将来学習用に、session log から curated dataset を export する。

## 背景

Reminder を最初は prompt-based で実装し、後から successful trace を使って SFT / LoRA するのが自然。
そのためには安全に scrub された dataset export が必要。

## 課題

- raw session log は学習データとして冗長かつ危険
- secret / absolute path / private repo 情報の scrub が必要
- success / failure trace を分けたい
- Reminder input/output 形式で抽出したい

## 提案する解決策

`llm-io.jsonl` 内の structured event を session_id + **turn_index** で join し、training dataset JSONL を export する。

**データソース設計（決定事項）**:
- `agent.reminder.completed` イベントから: `added_precautions_text` (出力テキスト配列), `feedback_kind`, `turn_index`, `session_id`, `task_at_call_time`, `precautions_at_call_time`, `feedback_excerpt`
- `agent.anvil_score.computed` イベントから: score フィールド（同 session_id + **turn_index** でマッチ）
- `chat.request`/`chat.response` イベントは**含めない**（原則）

**Reminder ログ拡張（必要）**:
- `agent.reminder.completed` ペイロードに追加するフィールド:
  - `task_at_call_time: String` — Reminder 呼び出し時の user_task
  - `precautions_at_call_time: Vec<String>` — 呼び出し時の active precautions テキスト（mask_secrets 適用済み）
  - `feedback_excerpt: String` — feedback excerpt（4 KiB cap + mask_secrets）
  - `added_precautions_text: Vec<String>` — 追加された precaution テキスト配列
  - `turn_index: usize` — セッション内の turn 番号（join キー）
- `agent.anvil_score.computed` ペイロードに `turn_index: usize` を追加

**実装方針**: `build_log_payload` に `inputs: Option<&ReminderInputs<'_>>` を追加し、Some の場合のみ追加フィールドをペイロードに含める（後方互換・変更範囲最小）。`ReminderInputs` には `active_precautions_at_call_time: &'a [String]`（`WorkingMemory.active_precautions` の Active エントリから text を直接取り出す。`format_for_prompt()` レンダリング結果を parse する方式は禁止）を追加する。

### 出力フォーマット

```json
{
  "input": {
    "task": "...",
    "active_precautions": [],
    "feedback_kind": "BashSandboxViolation",
    "feedback_excerpt": "..."
  },
  "output": {
    "added_precautions": []
  },
  "label": {
    "next_anvil_score": {
      "build_passed": true,
      "tests_passed": true,
      "user_visible_artifact": false
    },
    "success": true
  }
}
```

**フォーマット定義**:
- `input.repo_summary` は除外（現行ログに存在しない。将来拡張として別 Issue で対応）
- `label.success` の定義: `build_passed == Some(true) && tests_passed == Some(true)`（いずれかが None または false の場合は false）
- `prompt` フィールドは export に含めない（task/precautions と内容が重複するため）

### 機能

- success traces のみ export（`--success-only`）
- failed traces のみ export（`--failed-only`）
- Reminder training format のみ（Verifier format は将来拡張・別 Issue で対応）
- manual review 用 summary 生成（Markdown: `--output` 未指定で JSONL を stdout に出す場合は stderr、`--output <FILE>` 指定時は stdout。件数・success/failed 比率・scrub 件数・path 匿名化件数・corrupt line スキップ件数）

### CLI 仕様

```
anvil sessions export [OPTIONS]
```

オプション:
- `--output <FILE>`: 出力 JSONL ファイルパス（デフォルト: stdout）
- `--success-only`: success trace のみ（`--failed-only` と排他）
- `--failed-only`: failed trace のみ（`--success-only` と排他）
- `--all`: 全 workspace の session を対象（デフォルト: 現在 workspace のみ）
- `--session <ID>`: 単一 session 指定（`validate_session_dir` でパス検証）

`SessionsAction` enum に `Export` variant を追加（既存 `sessions` サブコマンド体系に統合）。

出力 stream は JSONL と summary を混在させない。`--output` 未指定時は JSONL を stdout、summary を stderr に出す。`--output <FILE>` 指定時は JSONL をファイル、summary を stdout に出す。

### 制約

- secret scrub: `mask_secrets` / `mask_payload_inplace` を SSOT として再利用
- absolute path anonymization: `anonymize_paths(text, work_root) -> String`（`replace_paths` → `mask_secrets` の順で適用）を `src/session/export.rs` に実装。dispatch の `workspace_root` を渡す。
- raw conversation は含めない（`chat.request`/`chat.response` イベントをスキップ）
- export は明示コマンドでのみ実行
- corrupt JSONL line は skip してカウント（`skipped_corrupt_count` として summary に表示）
- 1 セッションあたり最大 `MAX_EXPORT_JSONL_BYTES = 64 MiB` のファイルサイズキャップ

### データ join ロジック

`llm-io.jsonl` を BufReader + lines() で line-by-line streaming parse:
1. `agent.reminder.completed` を HashMap に蓄積（key: `(session_id, turn_index)`）
2. `agent.anvil_score.computed` を walk して同 `(session_id, turn_index)` でマッチ
3. マッチした reminder + score ペアを DatasetRecord として emit
4. マッチしない場合: `label.next_anvil_score = null`、`success = false`

`--session <ID>` 時: `validate_session_dir` でパス検証後、`state_root/sessions/<id>/llm-io.jsonl` を直接開く。
`--all` 時: `iter_session_dirs` で UUID ディレクトリを走査。

### 新規ファイル

- `src/session/export.rs`: export ロジック（JSONL reader、フィルタ、join、scrub pipeline）
  - 依存: `session` 層（`mask_secrets`, `discovery`）と `logging` 層（`mask_payload_inplace`）のみ。**`agent` 層への依存禁止。**

### 既存ファイルの変更範囲

- `src/cli.rs`: `SessionsAction::Export` variant 追加、`--success-only`/`--failed-only` 排他制約
- `src/session/mod.rs`: `pub mod export;` 追加
- `src/session/sessions_cli.rs`: `dispatch` に `Export` アーム追加
- `src/session/export.rs`（新規）: JSONL reader、フィルタ、join、scrub pipeline（scrub 統計は before/after 差分から field 単位でカウント。`mask_secrets`/`mask_payload_inplace` の既存 signature は変更しない）
- `src/agent/loop_run/reminder.rs`: `build_log_payload` 拡張（5 フィールド追加）、`ReminderInputs` に `active_precautions_at_call_time: &'a [String]` 追加、regression test 更新
- `src/agent/loop_run/turn.rs`: `agent.anvil_score.computed` ペイロードに `turn_index` 追加
- `CLAUDE.md`: `agent.reminder.completed` payload key count / SSOT regression test 説明と `agent.anvil_score.computed` payload フィールド説明を新仕様に更新

## 受け入れ条件

- [ ] JSONL export ができる（`anvil sessions export > out.jsonl`。リダイレクト先は summary を含まない pure JSONL）
- [ ] `--success-only` で success traces のみ export できる（`build_passed == Some(true) && tests_passed == Some(true)`）
- [ ] `--failed-only` で failed traces のみ export できる
- [ ] `--success-only` と `--failed-only` は排他（同時指定でエラー）
- [ ] secret-looking 値が除去される（mask_secrets + mask_payload_inplace 再利用）
- [ ] absolute path が匿名化される（`anonymize_paths` ヘルパー実装）
- [ ] manual review 用 summary が生成される（Markdown: `--output` 未指定時は stderr、`--output <FILE>` 指定時は stdout。件数・比率・scrub 統計・corrupt skip 件数）
- [ ] E2E テスト: `tests/dataset_export_smoke.rs` に closure DI で fixture jsonl を使ったテスト（Ollama 不要、最低 4 ケース: success/failed/reminder なし + corrupt line skip）
- [ ] stdout/stderr 分離の回帰テスト: `--output` 未指定時は stdout が pure JSONL（summary 混入なし）、`--output <FILE>` 指定時は JSONL がファイル・summary が stdout になることを検証するケースを含める
- [ ] `agent.reminder.completed` ペイロードに `task_at_call_time`, `precautions_at_call_time`, `feedback_excerpt`, `added_precautions_text`, `turn_index` を追加
- [ ] `agent.anvil_score.computed` ペイロードに `turn_index` を追加
- [ ] `payload_completed_has_all_keys` テストを新フィールド対応に更新
- [ ] `CLAUDE.md` のログペイロード契約説明を新フィールド対応に更新

## 親 Epic

- Epic F: Observability / Evaluation / Training Export (#449)

## 参照

- workspace/v0.1.1/feature.md (Issue #24, 論理番号)
- `src/session/case_record.rs`: CaseRecord 構造体（参照アーキテクチャ）
- `src/logging.rs`: mask_payload_inplace / is_secret_like_key SSOT
- `src/session/feedback.rs`: mask_secrets SSOT
- `src/agent/loop_run/reminder.rs`: agent.reminder.completed イベント定義（l.497-589）
- `src/agent/loop_run/turn.rs`: agent.anvil_score.computed イベント定義（l.3832-3839）
